### PR TITLE
Task h   entering additional payment details

### PR DESCRIPTION
**Pull request by adriancorcoran**
## Aim
Adds a dynamic form section using React

## How?
1. Add React to the application
1. Add a React-powered section to the form for the payment details
1. Add tests to check it's working

## Notes
Nope

### CODE REVIEW

File: app/javascript/packs/hello_react.jsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
+// Run this example by adding <%= javascript_pack_tag 'hello_react' %> to the head of your layout file,
```
Comment (Christoflurp) [inline app/javascript/packs/hello_react.jsx:1]: Could probably stand to delete this file, no?

File: app/views/orders/_order_form.html.erb [hunk 2]
```diff
@@ -7,3 +7,4 @@
     </div>
   </div>
 </div>
+<%= javascript_pack_tag("pay_type") %>
```
Comment (Christoflurp) [inline app/views/orders/_order_form.html.erb:10]: EOL

File: test/system/orders_test.rb [hunk 3]
```diff
@@ -3,47 +3,118 @@
...
-
-    fill_in "Address", with: @order.address
-    fill_in "Email", with: @order.email
-    fill_in "Name", with: @order.name
-    fill_in "Pay type", with: @order.pay_type
-    click_on "Update Order"
-
-    assert_text "Order was successfully updated"
-    click_on "Back"
+  test "check routing number is displayed" do
```
Comment (Christoflurp) [inline test/system/orders_test.rb:75]: Might be worth updating the title of this to `check check is displayed` (lol), or something similar to your naming convention on other pay types.

**Review by Christoflurp — APPROVED** [review]
Couple non-consequential thoughts as always :)

File: app/javascript/packs/hello_react.jsx (hunk 1 above)
Comment (adriancorcoran) [reply app/javascript/packs/hello_react.jsx:1]: Totally, held onto it to look back on as a starter example

File: test/system/orders_test.rb (hunk 3 above)
Comment (adriancorcoran) [reply test/system/orders_test.rb:75]: Nice one, missed that - thanks Chris :)

Comment (adriancorcoran) [reply test/system/orders_test.rb:75]: Nope, this one is fine, the option is `Check` however the fields that display are `Routing #` and `Account #` so gonna leave name as is

**Pull request merged**